mithril-aggregator/src/file_uploaders/gcp_uploader.rs

```diff
@@ -0,0 +1,138 @@
+use anyhow::{anyhow, Context};
+use async_trait::async_trait;
+use cloud_storage::{
+    bucket::Entity, bucket_access_control::Role, object_access_control::NewObjectAccessControl,
+    Client,
+};
+use slog::{info, Logger};
+use std::{env, path::Path};
+use tokio_util::codec::{BytesCodec, FramedRead};
+
+use mithril_common::{logging::LoggerExtensions, StdResult};
+
+use crate::{file_uploaders::FileUri, FileUploader};
+
+/// GcpUploader represents a Google Cloud Platform file uploader interactor
+pub struct GcpUploader {
+    bucket: String,
+    use_cdn_domain: bool,
+    logger: Logger,
+}
+
+impl GcpUploader {
+    /// GcpUploader factory
+    pub fn new(bucket: String, use_cdn_domain: bool, logger: Logger) -> Self {
+        Self {
+            bucket,
+            use_cdn_domain,
+            logger: logger.new_with_component_name::<Self>(),
+        }
+    }
+
+    fn get_location(&self, filename: &str) -> FileUri {
+        let mut uri = vec![];
+        if !self.use_cdn_domain {
+            uri.push("storage.googleapis.com");
+        }
+        uri.push(&self.bucket);
+        uri.push(filename);
+
+        FileUri(format!("https://{}", uri.join("/")))
+    }
+}
+
+#[async_trait]
+impl FileUploader for GcpUploader {
+    async fn upload(&self, filepath: &Path) -> StdResult<FileUri> {
+        if env::var("GOOGLE_APPLICATION_CREDENTIALS_JSON").is_err() {
```

Comment:
Maybe we will have to do this in the dependency builder in a future PR?

Reply:
Okay, that could be a good idea.